### PR TITLE
test(gateway): add tests for the `io.camunda.zeebe.util.jar.ExternalJarClassLoader` and resources in jar.

### LINKED ISSUE #8982: Test hypothesis that `ExternalJarClassLoader` can load resources from JAR
**Description**

I have such a problem: my interceptor cannot load resources placed in the JAR. It couldn't find it. Let's write some tests for it in the `io.camunda.zeebe.util.jar.ExternalJarClassLoaderTest`. Also, I would like that it would be backported in 1.2.x versions.

### PR DESCRIPTION
**Pull request by aivinog1**
## Description

I've decided to create this one to be sure that my resources will be properly loaded and could be accessible from the class loader.

## Related issues

closes #8982 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [X] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

### CODE REVIEW

File: util/src/test/java/io/camunda/zeebe/util/jar/ExternalJarClassLoaderTest.java [hunk 1]
```diff
@@ -52,4 +56,26 @@ void shouldUseSystemClassLoaderAsFallback(final @TempDir File tempDir)
...
+    final String resourceName = "test-resource.txt";
+    final File jarFile = new File(tempDir, "with-resource.jar");
+    try (final FileOutputStream fileOutputStream = new FileOutputStream(jarFile)) {
+      try (final JarOutputStream jarOutputStream = new JarOutputStream(fileOutputStream)) {
+        jarOutputStream.putNextEntry(new JarEntry(resourceName));
+        jarOutputStream.write(testResourceValue.getBytes(StandardCharsets.UTF_8));
+        jarOutputStream.closeEntry();
+      }
+    }
+    final var classLoader = ExternalJarClassLoader.ofPath(jarFile.toPath());
```
Comment (npepinpe) [inline util/src/test/java/io/camunda/zeebe/util/jar/ExternalJarClassLoaderTest.java:72]: :wrench: I would use var consistently. I think rule of thumb is use var when the type is obvious. So most of the declarations above, e.g. `final FileOutputStream fileOutputStream = new FileOutputStream`. In this test, all of them are obvious.

We don't have a clear rule though, so the only thing I would suggest is to stay consistent within a single code unit (either use var or don't :slightly_smiling_face:)

**Review by npepinpe — CHANGES_REQUESTED** [review]
:+1: 

Please have a look at the only comment and then we can merge

File: util/src/test/java/io/camunda/zeebe/util/jar/ExternalJarClassLoaderTest.java (hunk 1 above)
Comment (aivinog1) [reply util/src/test/java/io/camunda/zeebe/util/jar/ExternalJarClassLoaderTest.java:72]: @npepinpe Hey, it's done :)

**Comment by npepinpe** [comment]
bors merge

**Review by npepinpe — APPROVED** [review]
bors merge

ups :sweat_smile:

**Pull request merged**